staging/src/k8s.io/apiserver/pkg/server/filters/reqmgmt.go

```diff
@@ -159,84 +396,192 @@ func (tr *TypedConfigObjectReference) String() string {
 ) http.Handler {
-	rmStartOnce.Do(func() { rmSetup(loopbackClientConfig, serverConcurrencyLimit, requestWaitLimit) })
-
+	// reqMgmt := rmSetup(loopbackClientConfig, serverConcurrencyLimit, requestWaitLimit)
+	// --
+	pls := initPriorityLevelStates(priorityLevelConfigurations, requestWaitLimit)
+	rmState := RMState{
+		flowSchemas:            flowSchemas,
+		priorityLevelStates:    pls,
+		serverConcurrencyLimit: serverConcurrencyLimit,
+	}
+	// --
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		ctx := r.Context()
-		requestInfo, ok := apirequest.RequestInfoFrom(ctx)
-		if !ok {
-			handleError(w, r, fmt.Errorf("no RequestInfo found in context, handler chain must be wrong"))
-			return
-		}
+		// =======================
+		// TODO(aaron-prindle) CHANGE removed for early testing
+		// ctx := r.Context()
+		// requestInfo, ok := apirequest.RequestInfoFrom(ctx)
+		// if !ok {
+		// 	handleError(w, r, fmt.Errorf("no RequestInfo found in context, handler chain must be wrong"))
+		// 	return
+		// }
 
 		// Skip tracking long running events.
-		if longRunningRequestCheck != nil && longRunningRequestCheck(r, requestInfo) {
-			handler.ServeHTTP(w, r)
-			return
-		}
+		// if longRunningRequestCheck != nil && longRunningRequestCheck(r, requestInfo) {
+		// 	handler.ServeHTTP(w, r)
+		// 	return
+		// }
+
+		// get state of regMgmnt so we know all the obj/config values
+		// rmState := reqMgmt.curState.Load().(*RMState)
 
-		rmState := reqMgmt.curState.Load().(*RMState)
-		fs := reqMgmt.pickFlowSchema(r, rmState.flowSchemas, rmState.priorityLevelStates)
-		ps := reqMgmt.requestPriorityState(r, fs, rmState.priorityLevelStates)
+		// =======================
+
+		// analyze request to get flowSchema (so can get it's info - PriorityState, FlowDistinguisher)
+		// fs := reqMgmt.pickFlowSchema(r, rmState.flowSchemas, rmState.priorityLevelStates)
+
+		fs := pickFlowSchema(r, rmState.flowSchemas, rmState.priorityLevelStates)
+
+		// get the PriorityState, if exempt pass through
+		// ps := reqMgmt.requestPriorityState(r, fs, rmState.priorityLevelStates)
+		ps := requestPriorityState(r, fs, rmState.priorityLevelStates)
 		if *ps.config.Exempt {
 			klog.V(5).Infof("Serving %v without delay\n", r)
 			handler.ServeHTTP(w, r)
 			return
 		}
-		flowDistinguisher := reqMgmt.computeFlowDistinguisher(r, fs)
-		hashValue := reqMgmt.hashFlowID(fs.Name, flowDistinguisher)
+		// calculate the flow distinguisher for the flowSchema
+		// flowDistinguisher := reqMgmt.computeFlowDistinguisher(r, fs)
+		flowDistinguisher := computeFlowDistinguisher(r, fs)
+
+		// hash the flowDistinguisher (for shuffle sharding
+		// hashValue := reqMgmt.hashFlowID(fs.Name, flowDistinguisher)
+		hashValue := hashFlowID(fs.Name, flowDistinguisher)
+
+		// TODO(aaron-prindle) verify this is needed?
+		rmState.updateACV(ps) // TODO(aaron-prindle) rename updateACV?
+		ps.fqs.SetConcurrencyLimit(ps.concurrencyLimit)
```

Comment:
This does not need to be done at every request.  It needs to be done when the configuration changes, and during initial setup.  I do not understand why you removed the `concurrencyLimit` parameter from the FairQueueingSystem constructor.

Reply:
I originally did not understand why it would be set as I thought it was dynamic from rmState now I believe I understand the intention of concurrencyLimit.  This is updated, `concurrencyLimit` is again in the FairQueueingSystem constructor